plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -87,8 +86,10 @@ def __init__(self, repo, publish_conduit, config, distributor_type, association_
         self.add_child(PublishCompsStep())
         self.add_child(PublishMetadataStep())
         self.add_child(CloseRepoMetadataStep())
-        self.add_child(GenerateSqliteForRepoStep(self.get_working_dir()))
         self.add_child(RemoveOldRepodataStep())
+        self.add_child(GenerateSqliteForRepoStep(self.get_working_dir()))
+        if config.get_boolean('gpg_sign_metadata'):
```

Comment:
Signing used to happen as part of `CloseRemoMetadataStep` but `GenerateSqliteForRepoStep` was rewriting the repomd.xml afterwards and immediately invalidating the signature.  So we need to split the signing away from `CloseRepoMetadataStep` and make it a separate step that runs at the end.